### PR TITLE
Trigger comment trigger on linear app

### PR DESCRIPTION
**Pull request by bush4u**

### CODE REVIEW

**Comment by dylburger** [comment]
Thanks for submitting this PR! When we review PRs, we follow the [Pipedream component guidelines](https://pipedream.com/docs/components/guidelines/). If you're not familiar, here's a quick checklist:

- [ ] [Create components to address specific use cases whenever possible](https://pipedream.com/docs/components/guidelines/#component-scope)
- [ ] [Component `key`s should follow the format `app_name_slug-slugified-component-name`](https://pipedream.com/docs/components/guidelines/#component-key-pattern)
- [ ] Components should follow the [standard directory structure](https://pipedream.com/docs/components/guidelines/#folder-structure)
- [ ] [Prefer Node.js client libraries to REST APIs](https://pipedream.com/docs/components/guidelines/#using-apis-vs-client-libraries)
- [ ] When making API requests, [handle pagination](https://pipedream.com/docs/components/guidelines/#pagination) to ensure all data / events are processed
- [ ] [Use `secret` props](https://pipedream.com/docs/components/guidelines/#capturing-sensitive-data) to capture sensitive data
- [ ] Props and methods should be defined in [app files](https://pipedream.com/docs/components/guidelines/#app-files) whenever possible
- [ ] Document methods with [JS Docs](https://pipedream.com/docs/components/guidelines/#methods)
- [ ] [Use `optional` props](https://pipedream.com/docs/components/guidelines/#optional-vs-required-props) whenever possible, and [set a `default` value](https://pipedream.com/docs/components/guidelines/#default-values) where you can
- [ ] Use [async options](https://pipedream.com/docs/components/guidelines/#async-options) to accept user input wherever possible
- [ ] [Create subscription webhooks sources (vs polling sources) whenever possible](https://pipedream.com/docs/components/guidelines/#webhook-vs-polling-sources)
- [ ] [Always emit a `summary` with events](https://pipedream.com/docs/components/guidelines/#emit-a-summary)
- [ ] [Use a built-in deduping strategy wherever possible](https://pipedream.com/docs/components/guidelines/#deduping)
- [ ] When building webhook sources, [follow these best practices](https://pipedream.com/docs/components/guidelines/#webhook-sources)
- [ ] When building polling sources, [follow these best practices](https://pipedream.com/docs/components/guidelines/#polling-sources)

File: components/linear_app/sources/common/webhook.mjs [hunk 1]
```diff
@@ -31,7 +31,8 @@ export default {
       if (!this.getActions().includes(body?.action)) {
         return false;
       }
-      if (this.projectId) {
+      // projectId does not exist in the returned body data for Comment
+      if (this.projectId && !this.getResourceTypes().includes(constants.RESOURCE_TYPE.COMMENT)) {
```
Comment (andrewjschuang) [inline components/linear_app/sources/common/webhook.mjs:35]: Now that we have sources for issues and comments, would the function `isRelevant` better be implemented inside each individual source?

Comment (bush4u) [reply components/linear_app/sources/common/webhook.mjs]: Yes. Done

**Comment by DilanAthukorala** [comment]
@bush4u The comment created trigger is not getting filtered by Project Id param. 
image

**Comment by vunguyenhung** [comment]
Hello everyone, I have tested this PR and there're some test cases failed or needed improvement. 

Please check the test report below for more information
[Linear_App_3287.pdf](https://github.com/PipedreamHQ/pipedream/files/9041078/Linear_App_3287.pdf)

**Comment by vunguyenhung** [comment]
Hello everyone, I have tested this PR and there're some test cases failed or needed improvement. 

Please check the test report below for more information
[Linear_App_3287_2338.pdf](https://github.com/PipedreamHQ/pipedream/files/9056611/Linear_App_3287_2338.pdf)

**Comment by vunguyenhung** [comment]
Hi everyone, all test cases are passed! Ready for release! 

Test report
[Linear_App_3287_1732.pdf](https://github.com/PipedreamHQ/pipedream/files/9062739/Linear_App_3287_1732.pdf)

**Pull request merged**